src/pages/docs/creators/javascript/tutorials/rest-api-with-express.md

```diff
@@ -0,0 +1,278 @@
+    console.log('Connected to the Golem Network!')
+  })
+  .catch((err) => {
+    console.error('Error connecting to the Golem Network:', err)
+    process.exit(1)
+  })
+```
+
+Let's also add a handler for the `SIGINT` signal so that we can close the connection to the Golem Network when the app is stopped.
+
+```js
+process.on('SIGINT', async () => {
+  await golem.close()
+  process.exit(0)
+})
+```
+
+## Creating a retrievable task
+
+Let's add a new endpoint to our API that will take some text from request body and create a new task on the Golem Network. To do that, we will use the `createJob()` method. This method will give us a `Job` object that we can use to get the state of the job and it's results later. On the provider side we will run the `espeak` command to convert the text to speech, save it to a `.wav` file download that file to your local filesystem with the `downloadFile()` method. We will give the file a random name to avoid collisions.
+
+```js
+app.post('/tts', async (req, res) => {
+  if (!req.body) {
+    res.status(400).send('Missing text parameter')
+    return
+  }
+  const job = await golem.createJob(async (ctx) => {
+    const fileName = `${Math.random().toString(36).slice(2)}.wav`
+    await ctx
+      .beginBatch()
+      .run(`espeak "${req.body}" -w /golem/output/output.wav`)
+      .downloadFile('/golem/output/output.wav', `public/${fileName}`)
+      .end()
+    return fileName
+  })
+  res.send(`Job started! ID: ${job.id}`)
+})
+```
+
+## Getting the job state
+
+Now let's add another endpoint that will allow us to get the state of the job. We will use the `fetchState()` method of the `Job` object to get the state of the job and return it to the user.
+
+```js
+app.get('/tts/:id', async (req, res) => {
+  const job = golem.getJobById(req.params.id)
+  try {
+    const state = await job.fetchState()
+    res.send(state)
+  } catch (err) {
+    res.status(404).send('Job not found')
+  }
+})
+```
+
+## Getting the job results
+
+Finally, let's add an endpoint that will allow us to get the results of the job. We will use the `fetchResults()` method of the `Job` object to get the results of the job and return them to the user. We will also serve the files from the `/public` directory so that the user can easily listen to the results in a browser.
+
+```js
+// serve files in the /public directory
+app.use('/results', express.static('public'))
+
+app.get('/tts/:id/results', async (req, res) => {
+  const job = golem.getJobById(req.params.id)
+  try {
+    const results = await job.fetchResults()
+    res.send(
+      `Job completed successfully! Open the following link in your browser to listen to the result: http://localhost:${port}/results/${results}`
+    )
+  } catch (err) {
+    res.status(404).send('Job not found')
+  }
+})
+```
+
+## Running the API
+
+Start your app by running `node index.mjs` and run the following command in another terminal window (feel free to replace `Hello Golem` with any text you want):
```

Comment:
I think this whole section could be made more readable for the eyes.

> Start your app by running `node index.mjs` and run the following command in another terminal window (feel free to replace `Hello Golem` with any text you want):
> 
> ```bash
> curl \
>     --header "Content-Type: text/plain" \
>     --request POST \
>     --data "Hello Golem" \
>     http://localhost:3000/tts
> ```
> 
> You should see the following output:
> 
> ```bash
> Job started! ID: <job_id>
> ```
> 
> Now run (replace `<job_id>` with the job ID you got in the previous step):
> 
> ```bash
> curl http://localhost:3000/tts/<job_id>
> ```
> 





Instead of the above, I would go this route for the sake of clarity and less use of parantheses.






> ### Starting the App:
> Launch your application using:
> 
> ```bash
> node index.mjs
> ```
> 
> ### Sending a POST Request:
> In a separate terminal window, send a POST request. You can modify the message "Hello Golem" to any text of your choice:
> 
> ```bash
> curl \
>     --header "Content-Type: text/plain" \
>     --request POST \
>     --data "Hello Golem" \
>     http://localhost:3000/tts
> ```
> Upon executing the above, you should receive an output like:
> 
> ```bash
> Job started! ID: <job_id>
> ```
> 
> ### Fetching the Result:
> Use the returned job ID from the previous step and replace `<job_id>` in the command below:
> 
> ```bash
> curl http://localhost:3000/tts/<job_id>
> ```
> 
>